### PR TITLE
Fix adding new category

### PR DESCRIPTION
**Pull request by Hlavtox**
➡️ Note for QA and review - test are OK, the red is due to coveralls again.

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | My new function that automatically adds newly created categories to a default filter (added in https://github.com/PrestaShop/ps_facetedsearch/pull/728) does not work properly, because I forgot to call `$this->module->buildLayeredCategories();`. I use this oportunity to also clean up the related code a bit.
| Type?         | refacto
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | -
| How to test?  | Set up a default filter in faceted search. Go create a new category, add some product to it. Go to FO and see that you can see filters. Before, the category was added to the template, but the filters didn't work.

### Details
- Moved my adding function to Category class where it should be.
- Added the missing `$this->module->buildLayeredCategories();` call I forgot.
- Renamed `cleanAndRebuildCategoryFilters` to a more appropriate name `removeCategoryFromFilterTemplates` and passing only the category ID there, not the whole params array.

- - -
This change is [Reviewable](https://reviewable.io/reviews/prestashop/ps_facetedsearch/763)

### CODE REVIEW

**Review by SharakPL — APPROVED** [review]

**Review by SharakPL — DISMISSED** [review]

File: src/Hook/Category.php [hunk 1]
```diff
@@ -82,8 +84,8 @@ private function cleanAndRebuildCategoryFilters(array $params)
         foreach ($layeredFilterList as $layeredFilter) {
             $data = Tools::unSerialize($layeredFilter['filters']);
 
-            if (in_array((int) $params['category']->id, $data['categories'])) {
-                unset($data['categories'][array_search((int) $params['category']->id, $data['categories'])]);
+            if (in_array((int) $idCategory, $data['categories'])) {
+                unset($data['categories'][array_search((int) $idCategory, $data['categories'])]);
                 $this->database->execute(
                     'UPDATE `' . _DB_PREFIX_ . 'layered_filter`
                     SET `filters` = \'' . pSQL(serialize($data)) . '\',
```
Comment (kpodemski) [inline src/Hook/Category.php:91]:
```suggestion
                    SET `filters` = "' . pSQL(serialize($data)) . '"',
```

**Review by kpodemski — DISMISSED** [review]

**Comment by Hlavtox** [comment]
@kpodemski @SharakPL Changes applied guys. I unified the format of the two queries and added a flag in order not to generate the filters when nothing was changed. 👍 If it's OK, you can set Waiting for QA.

**Review by SharakPL — APPROVED** [review]

**Review by kpodemski — DISMISSED** [review]
For the next time I suggest using `DbQuery`, it makes things simpler 👍

**Comment by sallemiines** [comment]
Hello @Hlavtox 

We could'nt test this PR while there are some checks were not successful
Could you please resolve the conflict?

Thank you !

**Review by SharakPL — APPROVED** [review]

**Comment by Hlavtox** [comment]
@sallemiines Ines, do you read the PR description, please?

This is like a 10th PR for this module where this tests are failing, it's not an issue.

**Review by MhiriFaten — APPROVED** [review]
Hello @Hlavtox ,

I checked your PR and followed the steps mentioned 

1. Set up a default filter in faceted search
2. Create a new category
3. Add some product to it. 
4. Go to FO and see that you can see filters.

So it's QA approved :heavy_check_mark: 
Thanks.

**Comment by Hlavtox** [comment]
Thank you @MhiriFaten 🌹🌹🌹

**Pull request merged**